<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PenEditDialog</class>
 <widget class="QDialog" name="PenEditDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>400</width>
    <height>500</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Pen editor</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <property name="spacing">
    <number>4</number>
   </property>
   <property name="margin">
    <number>4</number>
   </property>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <property name="spacing">
      <number>4</number>
     </property>
     <item>
      <widget class="QLabel" name="resultLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>1</width>
         <height>1</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QWidget" name="propertiesWidget" native="true">
       <property name="minimumSize">
        <size>
         <width>230</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>230</width>
         <height>16777215</height>
        </size>
       </property>
       <layout class="QVBoxLayout" name="propertiesLayout">
        <property name="spacing">
         <number>4</number>
        </property>
        <property name="margin">
         <number>0</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <property name="spacing">
           <number>4</number>
          </property>
          <item>
           <widget class="QLabel" name="brushIconLabel">
            <property name="text">
             <string>icon</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="brushLabel">
            <property name="text">
             <string>Brush</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="brushButton">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_4">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="colorLayout">
          <property name="spacing">
           <number>4</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QLabel" name="colorLabel">
            <property name="text">
             <string>Color</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="widthLabel">
          <property name="text">
           <string>Width:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="widthSpinBox">
          <property name="decimals">
           <number>6</number>
          </property>
          <property name="maximum">
           <double>2147483647.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="styleLabel">
          <property name="text">
           <string>Style:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="styleComboBox">
          <item>
           <property name="text">
            <string>NoPen</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>SolidLine</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>DashLine</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>DotLine</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>DashDotLine</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>DashDotDotLine</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>CustomDashLine</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>MPenStyle</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="capStyleLabel">
          <property name="text">
           <string>Cap style:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="capStyleComboBox">
          <item>
           <property name="text">
            <string>FlatCap</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>SquareCap</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>RoundCap</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>MPenCapStyle</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="joinStyleLabel">
          <property name="text">
           <string>Join style:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="joinStyleComboBox">
          <item>
           <property name="text">
            <string>MiterJoin</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>BevelJoin</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>RoundJoin</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>SvgMiterJoin</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>MPenJoinStyle</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="cosmeticCheckBox">
          <property name="text">
           <string>Cosmetic</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="milerLimitLabel">
          <property name="text">
           <string>Miler limit:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="milerLimitSpinBox">
          <property name="decimals">
           <number>6</number>
          </property>
          <property name="maximum">
           <double>2147483647.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.010000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="dashOffsetLabel">
          <property name="text">
           <string>Dash offset:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="dashOffsetSpinBox">
          <property name="decimals">
           <number>6</number>
          </property>
          <property name="maximum">
           <double>2147483647.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="patternLabel">
          <property name="text">
           <string>Pattern:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QScrollArea" name="patternScrollArea">
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>228</width>
             <height>230</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout">
            <property name="spacing">
             <number>4</number>
            </property>
            <property name="margin">
             <number>4</number>
            </property>
            <item>
             <layout class="QVBoxLayout" name="patternLayout">
              <property name="spacing">
               <number>4</number>
              </property>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <item>
               <widget class="QToolButton" name="addButton">
                <property name="text">
                 <string/>
                </property>
                <property name="icon">
                 <iconset resource="../property.qrc">
                  <normaloff>:/objectcontroller/images/Add.png</normaloff>:/objectcontroller/images/Add.png</iconset>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_5">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
            <item>
             <spacer name="verticalSpacer">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>83</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <property name="spacing">
      <number>4</number>
     </property>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="okButton">
       <property name="text">
        <string>OK</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="cancelButton">
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_3">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../property.qrc"/>
 </resources>
 <connections/>
</ui>
